### PR TITLE
[IOTDB-2658] Generate logical plan for query statement

### PR DESCRIPTION
**Pull request by liuminghui233**
Generate logical plan for query statement ~

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/mpp/sql/planner/LogicalPlanner.java [hunk 1]
```diff
@@ -76,8 +84,155 @@ public LogicalPlanVisitor(Analysis analysis) {
...
+        Set<SourceNode> sourceNodes = planResultColumn(resultColumn);
+        for (SourceNode sourceNode : sourceNodes) {
+          String deviceName = sourceNode.getDeviceName();
+          deviceNameToSourceNodesMap
+              .computeIfAbsent(deviceName, k -> new HashSet<>())
+              .add(sourceNode);
+          deviceNameToDataRegionReplicaSetMap.computeIfAbsent(
+              deviceName,
+              k -> analysis.getDataPartitionInfo().getDataRegionReplicaSet(k, null).get(0));
+          sourceNode.setDataRegionReplicaSet(deviceNameToDataRegionReplicaSetMap.get(deviceName));
```
Comment (xingtanzjr) [inline server/src/main/java/org/apache/iotdb/db/mpp/sql/planner/LogicalPlanner.java:127]: Is it necessary to set DataRegionReplicaSet to SourceNode here ?

And... If there are several `DataRegionReplicaSet`s here, we only set one of them to the SourceNode, which is not correct

File: server/src/main/java/org/apache/iotdb/db/mpp/sql/planner/LogicalPlanner.java [hunk 2]
```diff
@@ -76,8 +84,155 @@ public LogicalPlanVisitor(Analysis analysis) {
...
+      }
+
+      planBuilder = planFilterNull(planBuilder, queryStatement.getFilterNullComponent());
+      planBuilder = planSort(planBuilder, queryStatement.getResultOrder());
+      planBuilder = planLimit(planBuilder, queryStatement.getRowLimit());
+      planBuilder = planOffset(planBuilder, queryStatement.getRowOffset());
+      return planBuilder.getRoot();
+    }
+
+    private PlanBuilder planSelectComponent(QueryStatement queryStatement) {
```
Comment (xingtanzjr) [inline server/src/main/java/org/apache/iotdb/db/mpp/sql/planner/LogicalPlanner.java:113]: Besides the series in ResultColumn, the related series in QueryFilter should also be considered here.

**Review by JackieTien97 — CHANGES_REQUESTED** [review]

File: server/src/main/java/org/apache/iotdb/db/mpp/sql/planner/plan/node/source/SeriesScanNode.java [hunk 3]
```diff
@@ -99,6 +99,16 @@ public void setDataRegionReplicaSet(DataRegionReplicaSet dataRegion) {
...
   }
 
+  @Override
+  public String getDeviceName() {
+    return seriesPath.getDevice();
+  }
+
+  @Override
+  protected String getExpressionString() {
+    return seriesPath.toString();
```
[the remark this answers is not in the document]
Comment (liuminghui233) [reply server/src/main/java/org/apache/iotdb/db/mpp/sql/planner/plan/node/source/SeriesScanNode.java:109]: fixed

File: server/src/main/java/org/apache/iotdb/db/mpp/sql/planner/LogicalPlanner.java (hunk 1 above)
Comment (liuminghui233) [reply server/src/main/java/org/apache/iotdb/db/mpp/sql/planner/LogicalPlanner.java:127]: Got it ~ I will remove it

File: server/src/main/java/org/apache/iotdb/db/mpp/sql/planner/LogicalPlanner.java [hunk 4]
```diff
@@ -76,8 +84,155 @@ public LogicalPlanVisitor(Analysis analysis) {
 
     @Override
     public PlanNode visitQuery(QueryStatement queryStatement, MPPQueryContext context) {
-      // TODO: Generate logical planNode tree for query statement
-      return null;
+      PlanBuilder planBuilder = planSelectComponent(queryStatement);
```
[the remark this answers is not in the document]
Comment (liuminghui233) [reply server/src/main/java/org/apache/iotdb/db/mpp/sql/planner/LogicalPlanner.java:87]: Indeed……if PlanBuilder still is meaningless later, I'll consider removing it.

File: server/src/main/java/org/apache/iotdb/db/mpp/sql/planner/LogicalPlanner.java (hunk 2 above)
Comment (xingtanzjr) [reply server/src/main/java/org/apache/iotdb/db/mpp/sql/planner/LogicalPlanner.java:112]: Sure, let's add a TODO here

Comment (liuminghui233) [reply server/src/main/java/org/apache/iotdb/db/mpp/sql/planner/LogicalPlanner.java:112]: added

**Pull request merged**